src/WebApp/Extensions/Extensions.cs

```diff
@@ -101,7 +99,7 @@ private static void AddAIServices(this IHostApplicationBuilder builder)
         if (builder.Configuration["OllamaEnabled"] is string ollamaEnabled && bool.Parse(ollamaEnabled))
         {
             builder.AddOllamaSharpChatClient("chat");
-            builder.Services.AddChatClient(b => b.GetRequiredService<IChatClient>())
+            builder.Services.AddChatClient(sp => (IChatClient)sp.GetRequiredKeyedService<IOllamaApiClient>("chat"))
```

Comment:
The Ollama component registers an IOllamaApiClient rather than an OllamaApiClient?

Reply:
We should update this to use the new API @aaronpowell

Reply:
Correct. That is pretty typical to register and use the interface (when available) instead of the concrete type. That way someone can easily mock it out in tests.

https://github.com/CommunityToolkit/Aspire/blob/3cda8176ea0f2026494df22514906dd4079880d7/src/CommunityToolkit.Aspire.OllamaSharp/AspireOllamaSharpExtensions.cs#L136-L148

Reply:
> We should update this to use the new API @aaronpowell

We will. See https://github.com/dotnet/eShop/pull/759#discussion_r1970579654. We just need the new API to be stable.